### PR TITLE
V5/users

### PR DESCRIPTION
**Pull request by Mh-Asmi**
This pull request makes the following changes:
- v5/users apis

Test checklist:
- [ ]

- [ ] I certify that I ran my checklist

Fixes ushahidi/platform# .

Ping @ushahidi/platform

### CODE REVIEW

File: src/Ushahidi/Modules/V5/Actions/User/Handlers/CreateUserCommandHandler.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+     * @return int
+     */
+    public function __invoke(Action $command) //: int
+    {
+        $this->isSupported($command);
+        $command->setId(
+            $this->user_repository->create($command->getEntity())
+        );
+    }
+}
```
Comment (kubabialy) [inline src/Ushahidi/Modules/V5/Actions/User/Handlers/CreateUserCommandHandler.php:42]: I think we agreed that for command handlers we can actually return the identifier. I'd rather have us doing an explicit return than mutate commands.

File: src/Ushahidi/Modules/V5/DTO/UserSearchFields.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * @var ?array
+     */
+    private $role;
+
+    public function __construct(Request $request)
+    {
+        $this->query = $request->query('q');
+        $this->role =
+            $request->query('role')
```
Comment (kubabialy) [inline src/Ushahidi/Modules/V5/DTO/UserSearchFields.php:23]:
```suggestion
        $this->role =  $request->query('role')
```

File: src/Ushahidi/Modules/V5/DTO/UserSearchFields.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        return $this->role;
+    }
+
+    public function roleAsMysqlInCondition(): ?string
+    {
+        $role_in_condition = "(";
+        foreach ($this->role as $key => $item) {
+            $role_in_condition .= $key==0 ? $item."," : $item;
+        }
+        return $role_in_condition.= ")";
```
Comment (kubabialy) [inline src/Ushahidi/Modules/V5/DTO/UserSearchFields.php:44]:
```suggestion
        return sprintf('(%s)', implode(',', $this->role));
```

File: src/Ushahidi/Modules/V5/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                "realname" => $request->input("realname"),
+                "role" => $request->input("role"),
+                "gravatar" => $request->input("gravatar"),
+                "logins" => 0,
+                "failed_attempts" => 0,
+                "last_login" => null,
+                "created" => time()
+            ]);
+        }
+        return ($user_entity);
```
Comment (kubabialy) [inline src/Ushahidi/Modules/V5/Http/Controllers/UserController.php:194]:
```suggestion
        if ($action === "update") {
            return new UserEntity([
                "id" => $user->id,
                "email" => $request->input("email", $user->email),
                "password" => $request->input("password", $user->email),
                "realname" => $request->input("realname", $user->email),
                "role" => $request->input("role", $user->email),
                "gravatar" => $request->input("gravatar", $user->email),
                "logins" => $request->input("logins", $user->email),
                "failed_attempts" => $request->input("failed_attempts", $user->email),
                "last_login" => $request->input("last_login", $user->email),
                "created" => $user->created,
                "updated" => time()
            ]);
         }
         return  new UserEntity([
                "email" => $request->input("email"),
                "password" => $request->input("password"),
                "realname" => $request->input("realname"),
                "role" => $request->input("role"),
                "gravatar" => $request->input("gravatar"),
                "logins" => 0,
                "failed_attempts" => 0,
                "last_login" => null,
                "created" => time()
            ]);
```

File: src/Ushahidi/Modules/V5/Http/Controllers/UserSettingController.php [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+                    $request->query('limit', FetchUserSettingQuery::DEFAULT_LIMIT),
+                    $request->query('page', 1),
+                    $request->query('sortBy', "id"),
+                    $request->query('order', FetchUserSettingQuery::DEFAULT_ORDER),
+                    []
+                    // $this->getSearchData($request->input(), FetchUserSettingQuery::AVAILABLE_SEARCH_FIELDS)
+                )
+            )
+        );
+        return $resourceCollection;
```
Comment (kubabialy) [inline src/Ushahidi/Modules/V5/Http/Controllers/UserSettingController.php:71]:
```suggestion
        return new UserSettingCollection(
            $this->queryBus->handle(
                new FetchUserSettingQuery(
                    $user_id,
                    $request->query('limit', FetchUserSettingQuery::DEFAULT_LIMIT),
                    $request->query('page', 1),
                    $request->query('sortBy', "id"),
                    $request->query('order', FetchUserSettingQuery::DEFAULT_ORDER),
                    []
                    // $this->getSearchData($request->input(), FetchUserSettingQuery::AVAILABLE_SEARCH_FIELDS)
                )
            )
        );
```

File: src/Ushahidi/Modules/V5/DTO/UserSearchFields.php (hunk 3 above)
Comment (Mh-Asmi) [reply src/Ushahidi/Modules/V5/DTO/UserSearchFields.php:44]: I don't need this function more , I forgot to remove it

File: src/Ushahidi/Modules/V5/Actions/User/Handlers/CreateUserCommandHandler.php (hunk 1 above)
Comment (Mh-Asmi) [reply src/Ushahidi/Modules/V5/Actions/User/Handlers/CreateUserCommandHandler.php:42]: As we discussed on slack we will update the command bus handler after completing the current merges then we will modify all command handlers to return the id.

**Review by kubabialy — APPROVED** [review]

**Pull request merged**